src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/ECDsaOpenSsl.cs

```diff
@@ -63,39 +57,25 @@ public ECDsaOpenSsl(IntPtr handle)
                 throw new ArgumentException(SR.Cryptography_OpenInvalidHandle, nameof(handle));
 
             ThrowIfNotSupported();
-            SafeEcKeyHandle ecKeyHandle = SafeEcKeyHandle.DuplicateHandle(handle);
-            _key = new ECOpenSsl(ecKeyHandle);
-            KeySizeValue = _key.KeySize;
+
+            using (SafeEcKeyHandle ecKeyHandle = SafeEcKeyHandle.DuplicateHandle(handle))
+            {
+                // CreateEvpPkeyFromEcKey already uprefs so nothing else to do
+                _key = new Lazy<SafeEvpPKeyHandle>(Interop.Crypto.CreateEvpPkeyFromEcKey(ecKeyHandle));
+            }
+
+            ForceSetKeySize(_key.Value.GetKeySizeBits());
         }
 
         /// <summary>
-        /// Obtain a SafeHandle version of an EVP_PKEY* which wraps an EC_KEY* equivalent
+        /// Obtain a SafeHandle version of an EVP_PKEY* equivalent
         /// to the current key for this instance.
         /// </summary>
-        /// <returns>A SafeHandle for the EC_KEY key in OpenSSL</returns>
+        /// <returns>A SafeHandle for the EVP_PKEY key in OpenSSL</returns>
         public SafeEvpPKeyHandle DuplicateKeyHandle()
```

Comment:
Are you changing this because it's no longer correct with newer versions of OpenSSL, or because it's just something you consider untidy?

Partially I'm asking because I don't know that docs changes for already-documented methods flow to dotnet/docs; so you'll have to mirror this/these update(s).

If someone can always pass this SafeHandle down to EVP_PKEY_get0_EC_KEY and it would succeed, then the docs are correct (enough).  If keys opened from a named provider don't guarantee that, then the docs are too specific and need to change.

Reply:
I guess it was correct enough but it felt slightly off therefore the change, will revert